core/app/[locale]/(default)/cart/_actions/add-shipping-info.ts

```diff
@@ -0,0 +1,126 @@
+'use server';
+
+import { revalidateTag } from 'next/cache';
+
+import { getSessionCustomerAccessToken } from '~/auth';
+import { client } from '~/client';
+import { graphql } from '~/client/graphql';
+import { TAGS } from '~/client/tags';
+
+const UpdateCheckoutShippingConsignmentMutation = graphql(`
+  mutation UpdateCheckoutShippingConsignment($input: UpdateCheckoutShippingConsignmentInput!) {
+    checkout {
+      updateCheckoutShippingConsignment(input: $input) {
+        checkout {
+          entityId
+          shippingConsignments {
+            availableShippingOptions {
+              cost {
+                value
+              }
+              description
+              entityId
+            }
+          }
+        }
+      }
+    }
+  }
+`);
+
+const AddCheckoutShippingConsignmentsMutation = graphql(`
+  mutation AddCheckoutShippingConsignments($input: AddCheckoutShippingConsignmentsInput!) {
+    checkout {
+      addCheckoutShippingConsignments(input: $input) {
+        checkout {
+          entityId
+          shippingConsignments {
+            availableShippingOptions {
+              cost {
+                value
+              }
+              description
+              entityId
+            }
+          }
+        }
+      }
+    }
+  }
+`);
+
+interface Props {
+  checkoutEntityId: string;
+  shippingId?: string;
+  address: {
+    countryCode: string;
+    city?: string;
+    stateOrProvince?: string;
+    postalCode?: string;
+  };
+  lineItems: Array<{ quantity: number; lineItemEntityId: string }>;
+}
+
+export const addShippingInfo = async ({
+  checkoutEntityId,
+  shippingId,
+  address,
+  lineItems,
+}: Props) => {
+  const customerAccessToken = await getSessionCustomerAccessToken();
+
+  let response;
+
+  if (shippingId) {
+    response = await client.fetch({
```

Comment:
:nit: Can we do this without the reassignment? e.g. `const updateResponse = await ....`

Reply:
@jorgemoya I concur here, I think avoiding `let` whenever possible is a good idea 😄

Reply:
We could just add `const response` in the `if` block with the early return, and just use another `const response` afterwards

Reply:
@jorgemoya What if we just split it into separate functions like

```
async function updateCheckoutShippingConsignment(input): CheckoutResult {
  // ... (no revalidateTag needed)
  return response.data.checkout.updateCheckoutShippingConsignment?.checkout;
} 

async function addCheckoutShippingConsignments(input): CheckoutResult {
  // ... (no revalidateTag needed)
  return response.data.checkout.updateCheckoutShippingConsignment?.checkout;
} 

export const addShippingInfo = async ({ ... }) => {

  // ...

  const response = shippingId
    ? await updateCheckoutShippingConsignment(input)
    : await addCheckoutShippingConsignments(input);
    
  revalidateTag(TAGS.checkout);
  
  return result;
};
```

Reply:
Done!